src/main/java/uk/gov/pay/directdebit/mandate/services/gocardless/GoCardlessMandateStateCalculator.java

```diff
@@ -35,8 +39,20 @@
         this.goCardlessEventDao = goCardlessEventDao;
     }
 
-    Optional<DirectDebitStateWithDetails<MandateState>> calculate(GoCardlessMandateIdAndOrganisationId goCardlessMandateIdAndOrganisationId) {
-        return goCardlessEventDao.findLatestApplicableEventForMandate(goCardlessMandateIdAndOrganisationId, GOCARDLESS_ACTIONS_THAT_CHANGE_STATE)
+    Optional<DirectDebitStateWithDetails<MandateState>> calculate(Mandate mandate) {
+        Optional<GoCardlessEvent> latestGoCardlessEvent = mandate.getPaymentProviderMandateId()
+                .map(paymentProviderMandateId -> {
+                    GoCardlessOrganisationId goCardlessOrganisationId = mandate.getGatewayAccount().getOrganisation()
+                            .orElseThrow(() -> new GatewayAccountMissingOrganisationIdException(mandate.getGatewayAccount()));
+
+                    return goCardlessEventDao.findLatestApplicableEventForMandate(
+                            (GoCardlessMandateId) paymentProviderMandateId,
```

Comment:
Shame we don’t have a generic type parameter on `Mandate` indicating the type of `PaymentProviderMandateId` so we don’t need casts like this. But that might make code that deals with a mandate without knowing what provider it is more painful. So maybe not.